src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineWrappingInBlockCheck.java

```diff
@@ -0,0 +1,440 @@
+                verifyTop(leftBrace);
+                verifyBottom(rightBrace);
+            }
+        }
+        else {
+            // For the OBJBLOCK/SWITCH the left curly brace is the first child and
+            // the right curly brace is the last child of their common parent.
+            final DetailAST rightBrace = leftBrace.getParent().getLastChild();
+            // This check does not validate empty blocks.
+            if (leftBrace.getNextSibling() != rightBrace) {
+                verifyTop(leftBrace);
+                verifyBottom(rightBrace);
+            }
+        }
+    }
+
+    /**
+     * Verifies that a specified code block matches the policy for the top part of the block.
+     * @param brace token for left curly brace
+     */
+    private void verifyTop(DetailAST brace) {
+
+        if (topSeparator != BlockWrapOption.EMPTY_LINE_ALLOWED) {
+            final DetailAST next;
+            if (isRightCurlyChildOf(brace)) {
+                next = brace.getFirstChild();
+            }
+            else {
+                next = brace.getNextSibling();
+            }
+            int firstLine = next.getLineNo();
+            DetailAST child = next.getFirstChild();
+            // Finding the topmost line of an AST is a bit tricky. Usually, this is the first line
+            // of the corresponding comment, but in AST there can be an empty MODIFIERS token.
+            // In this case, the comment will be attached to the TYPE token.
+            while (child != null) {
+                firstLine = Math.min(firstLine, child.getLineNo());
+                DetailAST candidate = child.getFirstChild();
+                if (candidate == null) {
+                    candidate = child.getNextSibling();
+                }
+                child = candidate;
+            }
+            final int lines = firstLine - brace.getLineNo();
+            if (topSeparator == BlockWrapOption.EMPTY_LINE && lines <= 1) {
+                log(brace, MSG_SHOULD_BE_EMPTY_LINE_AT_TOP, brace.getText());
+            }
+            else if (topSeparator == BlockWrapOption.NO_EMPTY_LINE && lines > 1) {
+                log(brace, MSG_SHOULD_BE_NO_EMPTY_LINE_AT_TOP, brace.getText());
+            }
+        }
+    }
+
+    /**
+     * Verifies that a specified code block matches the policy for the bottom part of the block.
+     * @param brace token for right curly brace
+     */
+    private void verifyBottom(DetailAST brace) {
+
+        if (bottomSeparator != BlockWrapOption.EMPTY_LINE_ALLOWED) {
+            final DetailAST previous = brace.getPreviousSibling();
+            int lastLine = previous.getLineNo();
+            DetailAST child = previous.getLastChild();
+            while (child != null) {
+                lastLine = Math.max(lastLine, child.getLineNo());
+                child = child.getLastChild();
+            }
+            final int lines = brace.getLineNo() - lastLine;
```

Comment:
all lines above sounds like a static method to get lines count.

Reply:
This code has already been removed and replaced with a much faster one.